### ISSUE TITLE
Wrong training steps when images are in jxl format

### ISSUE THREAD

**Issue by Hirmuolio** [kohya-ss/sd-scripts#2087]
When training images are in jxl format the script ends up using wrong number of epochs and step count.

Example: 500 images, 10 epochs, batch size 2. Identical training settings.

With jxl the script thinks it should train for 7 epochs and 1600 steps instead.
With png it correctly does 10 epochs and 2500 steps.

The training still seems to work with jxl. Something resembling the training images was learned.

Using bmaltais GUI.

Parts of the console output:

With jxl:

09:01:51-937289 INFO     Folder 1_all: 1 repeats found  
09:01:51-937289 INFO     Folder 1_all: 0 images found  
09:01:51-937289 INFO     Folder 1_all: 0 * 1 = 0 steps
09:01:51-937289 INFO     Regularization factor: 1
09:01:51-937289 INFO     Train batch size: 2
09:01:51-937289 INFO     Gradient accumulation steps: 1
09:01:51-937289 INFO     Epoch: 10

09:01:51-937289 INFO     max_train_steps (0 / 2 / 1 * 10 * 1) = 0
INFO     set image size from cache files: 500/500
INFO     found directory D:\sd\test\1_all contains 500 image files
INFO     500 train images with repeats.

num train images * repeats / 学習画像の数×繰り返し回数: 500
num validation images * repeats / 学習画像の数×繰り返し回数: 0
num reg images / 正則化画像の数: 0
num batches per epoch / 1epochのバッチ数: 252
num epochs / epoch数: 7
batch size per device / バッチサイズ: 2
gradient accumulation steps / 勾配を合計するステップ数 = 1
total optimization steps / 学習ステップ数: 1600

With png:

09:04:28-150206 INFO     Folder 1_all: 1 repeats found
09:04:28-150206 INFO     Folder 1_all: 500 images found
09:04:28-150206 INFO     Folder 1_all: 500 * 1 = 500 steps
09:04:28-150206 INFO     Regularization factor: 1
09:04:28-150206 INFO     Train batch size: 2
09:04:28-150206 INFO     Gradient accumulation steps: 1
09:04:28-150206 INFO     Epoch: 10
09:04:28-150206 INFO     max_train_steps (500 / 2 / 1 * 10 * 1) = 2500

INFO     set image size from cache files: 500/500
INFO     found directory E:\training\quartet3\1_all contains 500 image files
INFO     500 train images with repeats.

num train images * repeats / 学習画像の数×繰り返し回数: 500
num validation images * repeats / 学習画像の数×繰り返し回数: 0
num reg images / 正則化画像の数: 0
num batches per epoch / 1epochのバッチ数: 252
num epochs / epoch数: 10
batch size per device / バッチサイズ: 2
gradient accumulation steps / 勾配を合計するステップ数 = 1
total optimization steps / 学習ステップ数: 2500

**Comment by Hirmuolio** [comment]
Yeah seems to be problem with gui.
